pytest_splunk_addon/splunk.py

```diff
@@ -800,6 +802,50 @@ def file_system_prerequisite():
         os.mkdir(monitor_dir)
 
 
+@pytest.fixture(scope="session")
+def splunk_dm_recommended_fields(splunk_search_util):
+    recommended_fields = defaultdict(list)
+
+    def _find(name, dictionary):
+        for key, value in dictionary.items():
+            if key == name:
+                yield value
+            elif isinstance(value, dict):
+                yield from _find(name, value)
+            elif isinstance(value, list):
+                for element in value:
+                    if isinstance(element, dict):
+                        yield from _find(name, element)
+
+    def update_recommended_fields(model, datasets):
+        model_key = f"{model}:{':'.join(datasets)}".strip(":")
+        if model_key not in recommended_fields:
+            for cim_model in splunk_search_util.getFieldValuesList(
+                f'| rest /servicesNS/-/-/data/models/{model} | fields "eai:data"'
```

Comment:
Can you please update the description of this PR? I know what you are trying to do here, but may be confusing for the people who don't.

Reply:
yep,
this is rare situation where I do agree that comment would make sense

Reply:
BTW:
Checking the code below I see a functionality I was developing some time back when was dealing with CIM
What about extracting CIM related functionality do dedicated library?

@artemrys

Reply:
Description added